src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6322,12 +6358,182 @@ private IOperation VisitNoneOperationExpression(IOperation operation)
 
         public override IOperation? VisitInterpolatedStringHandlerCreation(IInterpolatedStringHandlerCreationOperation operation, int? captureIdForResult)
         {
-            return VisitNoneOperation(operation, captureIdForResult);
+            // We turn the interpolated string into a call to create the handler type, a series of append calls (potentially with branches, depending on the
+            // handler semantics), and then evaluate to the handler flow capture temp.
+
+            SpillEvalStack();
+            RegionBuilder resultRegion = CurrentRegionRequired;
+            var handlerCaptureId = captureIdForResult ?? GetNextCaptureId(resultRegion);
+
+            var constructorRegion = new RegionBuilder(ControlFlowRegionKind.LocalLifetime);
+            EnterRegion(constructorRegion);
+
+            BasicBlockBuilder? resultBlock = null;
+            if (operation.HandlerCreationHasSuccessParameter || operation.HandlerAppendCallsReturnBool)
+            {
+                resultBlock = new BasicBlockBuilder(BasicBlockKind.Block);
+            }
+
+            // Any placeholders for arguments should have already been created, except for the out parameter if it exists.
+            int outParameterFlowCapture = -1;
+            IOperation? outParameterPlaceholder = null;
+
+            if (operation.HandlerCreationHasSuccessParameter)
```

Comment:
>if (operation.HandlerCreationHasSuccessParameter) [](http://example.com/codeflow?start=12&length=49)

We can call ```collectAppendCalls``` before this ```if``` and, if the set is empty, not enter the ```if```, leaving ```outParameterFlowCapture``` == -1. Then we suppress creation of conditional branch in the ```if``` below when ```outParameterFlowCapture == -1```. Then we can adjust ```VisitInterpolatedStringHandlerArgumentPlaceholder``` to create ```IDiscardOperation``` in this case. This way we can eliminate an unnecessary branch, see ```InterpolatedStringHandlerConversionFlow_16``` for example. This work doesn't look prohibiteviley complex or expensive to me.

Reply:
I filed [https://github.com/dotnet/roslyn/issues/57611](https://github.com/dotnet/roslyn/issues/57611) to track doing this work at some point in the future.